docs/source/examples/VOC2012Segmentation.rst

```diff
@@ -0,0 +1,197 @@
+which is typically stored in a json file.
+
+.. literalinclude:: ../../../tensorbay/opendataset/VOC2012Segmentation/catalog.json
+   :language: json
+   :name: voc2012segmentation-catalog
+   :linenos:
+
+The annotation types for "VOC2012 Segmentation" are :ref:`reference/label_format/SemanticMask:SemanticMask`
+and :ref:`reference/label_format/InstanceMask:InstanceMask`, and there are 22
+:ref:`reference/label_format/CommonLabelProperties:Category` types for :ref:`reference/label_format/SemanticMask:SemanticMask`. There are 2
+:ref:`reference/label_format/CommonLabelProperties:Category` types for :ref:`reference/label_format/InstanceMask:InstanceMask`, category 0 represents the
+background, and category 255 represents the border of instances.
+
+.. note::
+
+    The categories in instance subcatalog are for pixel value which are not instance ids.
+
+.. important::
+
+   See :ref:`catalog table <reference/dataset_structure:catalog>` for more catalogs with different label types.
+
+Step 2: Write the Dataloader
+============================
+
+A :ref:`reference/glossary:dataloader` is needed to organize the dataset into
+a :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+.. literalinclude:: ../../../tensorbay/opendataset/VOC2012Segmentation/loader.py
+   :language: python
+   :name: voc2012segmentation-dataloader
+   :linenos:
+
+See :ref:`SemanticMask annotation <reference/label_format/SemanticMask:SemanticMask>`
+and :ref:`InstanceMask annotation <reference/label_format/InstanceMask:InstanceMask>` for more details.
+
+.. note::
+
+   Since the :ref:`VOC2012 Segmentation dataloader <voc2012segmentation-dataloader>` above is already included in
+   TensorBay, so it uses relative import.
+   However, the regular import should be used when writing a new dataloader.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Organize Dataset / regular import"""
+   :end-at: from tensorbay.label import InstanceMask, SemanticMask
+
+There are already a number of dataloaders in TensorBay SDK provided by the community.
+Thus, instead of writing, importing an available dataloader is also feasible.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Organize dataset / import dataloader"""
+   :end-before: """"""
+
+.. note::
+
+   Note that catalogs are automatically loaded in available dataloaders, users do not have to write them again.
+
+.. important::
+
+   See :ref:`dataloader table <reference/glossary:dataloader>` for dataloaders with different label types.
+
+*******************
+ Visualize Dataset
+*******************
+
+Optionally, the organized dataset can be visualized by **Pharos**, which is a TensorBay SDK plug-in.
+This step can help users to check whether the dataset is correctly organized.
+Please see :ref:`features/visualization:Visualization` for more details.
+
```

Comment:
The mask type of label is not supported in pharos yet. Remove this section for now and something like `See the visualization on TenosrBay website` after `upload dataset`?